In [50]:
# bibliotecas necessárias
import pandas as pd

# criando objeto com caminho e carregando arquivo csv
path = '/home/robson/repositorios/bike-manufacturing-company/data-lake/landing/Sales.Customer.csv'
loaded_csv = pd.read_csv(path, sep=';')

# criando dataframe pandas
df = pd.DataFrame(loaded_csv)

#### Análise estrutural

In [51]:
# verificando estrutura do dataframe (rows, columns)
df.shape

(19820, 7)

In [52]:
# verificando tipos de dados
df.dtypes

CustomerID         int64
PersonID         float64
StoreID          float64
TerritoryID        int64
AccountNumber     object
rowguid           object
ModifiedDate      object
dtype: object

In [53]:
# analisando valores unicos de cada coluna nao-numerica
df.describe(exclude='number')

AccountNumber                               rowguid  \
count          19820                                 19820   
unique         19820                                 19820   
top       AW00000001  3F5AE95E-B87D-4AED-95B4-C3797AFCB74F   
freq               1                                     1   

                   ModifiedDate  
count                     19820  
unique                        1  
top     2014-09-12 11:15:07.263  
freq                      19820

#### Análise qualitativa

In [54]:
# verificando existencia de dados nulos por coluna
df.isnull().sum()

CustomerID           0
PersonID           701
StoreID          18484
TerritoryID          0
AccountNumber        0
rowguid              0
ModifiedDate         0
dtype: int64

### Conclusão:

1- PersonID possui 701 campos vazios. No caso, é prudente excluir apenas as linhas com tais dados em branco.

2- StoreID possue 18484 campos vazios. No caso, como não será feita nenhuma análise que necessite de identificador de loja, a coluna será eliminada.

3- As colunas CustomerID, PersonID, TerritoryID devem ser modificadas para o tipo string.